source/dotnet/Library/AdaptiveCards.Rendering.Html/AdaptiveCardRenderer.cs

```diff
@@ -477,40 +482,74 @@ protected static void AddContainerElements(HtmlTag uiContainer, IList<AdaptiveEl
                             }
                         }
 
+                        if (!cardElement.IsVisible)
+                        {
+                            uiElement.Style("display", "none");
+
+                            if (uiSeparator != null)
+                            {
+                                uiSeparator.Style("display", "none");
+                            }
+                        }
+                        else
+                        {
+                            // if it's visible and it's the first element, hide the separator
+                            if (isFirstElement)
+                            {
+                                if (uiSeparator != null)
+                                {
+                                    uiSeparator.Style("display", "none");
+                                }
+
+                                isFirstElement = false;
+                            }
+                        }
+
+                        if (!String.IsNullOrEmpty(cardElement.Id))
```

Comment:
use IsNullOrWhiteSpace rather than IsNullOrEmpty

Reply:
btw wrt our offline discussion, you can't call isNullOrWhitespace on the instance/object directly unless there is an extension method defined (probably would be neat) so incorrect on that front also; btw prefer using the string alias instead of the String class name generally but for now consistency with rest of the code base is fine!

---
In reply to: [280228354](https://github.com/Microsoft/AdaptiveCards/pull/2767#discussion_r280228354) [](ancestors = 280228354)